samcli/lib/package/artifact_exporter.py

```diff
@@ -265,3 +266,47 @@ def delete(self):
                     # Delete code resources
                     exporter = exporter_class(self.uploaders, None)
                     exporter.delete(resource_id, resource_dict)
+
+    def get_s3_info(self):
+        """
+        Iterates the template_dict resources with S3 EXPORT_DESTINATION to get the
+        s3_bucket and s3_prefix information for the purpose of deletion.
+        Method finds the first resource with s3 information, extracts the information
+        and then terminates.
+        """
```

Comment:
I think we should also mention it is safe to assume all packaged files are in the same S3 bucket and with the same prefix, because this is how `sam package` and `sam deploy` works now.

Reply:
That is a good point, this logic would only work for stacks which are created by sam. In the case we get bucket information which is not correct, SAM will not be able to tell that it is wrong and the template file wont be deleted.